### PR TITLE
Financial Metrices Capstone Project

### PR DESCRIPTION
**Pull request by omarsalem7**
## What does this PR do?
This PR is raised to capstone project Financial app
## Description of the task to be completed:

- [ ] created a react app using npx create-react-app 
- [ ] created a development branch and made it the default.
- [ ] created assets folder for the images.
- [ ] created components folder for the reusable components.
- [ ] created a redux folder for the store.
- [ ] created styles folder for all CSS files.
- [ ] created pages folder for the structure of the pages.
- [ ] Wrote tests 
- [ ] Added a 404 error page
- [ ] Added a loader to help keep users waiting while data is being fetched
- [ ] Added dynamic routing the details page
- [ ] Implemented search bar functionality using reducers and redux store
- [ ] Implemented charts to present the dynamically generated net income and gross profit in a line chart and bar chart.

## Presentation video
[video](https://www.loom.com/share/69b19f9247634dfebb80c565dd43c8e3)
## See Presentation GIF
screenshot

### CODE REVIEW

File: src/components/LineChart.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+// import Chart from 'chart.js/auto';
+import PropTypes from 'prop-types';
+import { Line } from 'react-chartjs-2';
+
+const LineChart = ({ result }) => {
+  // console.log(Chart);
```
Comment (MahmoudBakr23) [inline src/components/LineChart.js:7]: - For cleaner code, please remove unnecessary code lines! 🙏

File: src/pages/StockDetailPage.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    );
+  }
+
+  if (loading) {
+    return (
+      <>
+        <Spinner />
+      </>
+    );
+  }
```
Comment (MahmoudBakr23) [inline src/pages/StockDetailPage.js:40]: - I see you've used this logic across multiple components to decide whether the component will display the load page or the error page before rendering. I have a suggestion to save it in a separate function in some file `./loadCondition' and then reuse it whenever you want. This makes less and cleaner code!

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @omarsalem7 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

You aced this! 🥇 Keep going 🚀

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omarsalem7** [comment]
Good Afternoon @MahmoudBakr23 👋
Thanks for your feedback 🙏
Have a nice day 😄

**Pull request merged**